src/entry.js

```diff
@@ -43,23 +43,27 @@ class Entry {
       clock: new Clock(clockId, clockTime),
     }
 
-    // If signing key was passedd, sign the enrty
-    if (keystore && signKey) {
-      entry = await Entry.signEntry(keystore, entry, signKey) 
+    // If acl was passed, sign the entry (=authorize)
+    if (acl) {
+      entry = await Entry.sign(entry, acl) 
     }
 
     entry.hash = await Entry.toMultihash(ipfs, entry)
     return entry
   }
 
-  static async signEntry (keystore, entry, key) {
-    const signature = await keystore.sign(key, Buffer.from(JSON.stringify(entry)))
+  static async sign (entry, acl) {
```

Comment:
A smart contract access controller will want to augment the entry with a `chainSignature` and `chainPublicKey`, or something like that.